src/main.rs

```diff
@@ -65,7 +65,19 @@ async fn main() -> Result<(), Error> {
     launch_info();
 
     use log::LevelFilter as LF;
-    let level = LF::from_str(&CONFIG.log_level()).expect("Valid log level");
+    let level = LF::from_str(&CONFIG.log_level()).unwrap_or_else(|_| {
+        let mut valid_log_levels = String::new();
+        LF::iter().fold(true, |first, elem| {
+            let mut joinstr = ", ".to_string();
+            if first {
+                joinstr = String::new();
+            }
+            valid_log_levels = format!("{}{}{}", valid_log_levels, joinstr, elem.as_str().to_lowercase());
+            false
+        });
```

Comment:
I think this can be simplified:
```suggestion
        let valid_log_levels = LF::iter().map(|lvl| lvl.as_str().to_lowercase()).collect::<Vec<String>>().join(", ");
```

Reply:
awesome, thanks. this is why it's awesome to have Rust experts around!